tests/runner.py

```diff
@@ -426,11 +426,17 @@ def setUpClass(cls):
   def setUp(self):
     super(RunnerCore, self).setUp()
     self.settings_mods = {}
-    self.emcc_args = ['-Werror']
+    self.emcc_args = []
     self.save_dir = EMTEST_SAVE_DIR
     self.env = {}
     self.temp_files_before_run = []
 
+    # Ideally all our tests code would be built with -Wall, -Werror and -s STRICT.  For now
+    # we just enable a subset of the options that STRICT would enable.
+    self.emcc_args.append('-Werror')
+    self.set_setting('LLD_REPORT_UNDEFINED', 1)
+    self.set_setting('IGNORE_MISSING_MAIN', 0)
```

Comment:
do we have enough tests that check the default values of these flags (which is what our users use mostly)? I guess we have `other`, maybe that's enough.

Reply:
In general for each setting we have a test that test the effect of it both enabled and disabled.  So I think we should be fine.  But you raise a good point.  I think maybe this is the wrong approach.   Maybe we should consider this one of the "modes" in which we run the test suite.   I'll revert his part of the change for now.

(BTW, I would like to see all tests (including other) use this system and honor these default flags uses by the test suite)

Reply:
Another option is to move this test to `other` if you want to control all the commands 100% manually. `core` is useful for things that you trust the default core modes to cover, as I see it.